cpp/tests/utilities/base_fixture.hpp

```diff
@@ -116,7 +138,8 @@ class UniformRandomGenerator {
 
  private:
   uniform_distribution dist{};         ///< Distribution
-  Engine rng{std::random_device{}()};  ///< Random generator
+  
+  Engine rng{std::mt19937_64{detail::random_generator_incrementing_seed()}()};    ///< Random generator
```

Comment:
Instead of using this logic for all users of the random generator, make the seed a constructor argument and put this function call at the point of construction in the binops tests.

Reply:
I assume you mean a constructor argument with a reasonable default value?

Reply:
I don't think it's wise to default this behavior if we want determinism because now the random seed is dependent on the order in which the tests are executed. 

Also, why not just make the seed a static member of the class instead of defining a whole new function and TU?

Reply:
If it were a static member, the fact that it's templated would kind of obfuscate things.  Better to have a single place that generates a seed rather than a place per type.

I updated the PR with your change for making the seed a constructor parameter.  I'm definitely not a fan of making it a required parameter with no default.  If you look at the changes, it's just a bunch of places where people want to declare a min and max value - why should they have to provide a seed if there's a good default alternative?  The puts a little bit more of a knowledge burden on the user.

Reply:
Now that some time has passed since my original review, I guess making this the default behavior is fine. We're trading what used to be a random seed that the user was ignorant of for a deterministic seed that the user is still ignorant of. I'd still like to see a way to explicitly provide a seed if the user wants, but defaulting to the counting seed is fine.

Reply:
We can just make the seed parameter in the constructor default to using the generator.  User can override if they want.